indexer/src/lib.rs

```diff
@@ -130,7 +132,7 @@ impl<'a, INDEX: Index> Indexer<'a, INDEX> {
                                             EventType::Put => {
                                                 match self.storage.get_for_event(&data).await {
                                                     Ok((k, data)) => {
-                                                        if let Err(e) = self.index_doc(self.index.index(), writer.as_mut().unwrap(), &k, &data).await {
+                                                        if let Err(e) = self.index_doc(self.index.index(), writer.as_mut().unwrap(), &k, &data, &mut indexed).await {
                                                             log::warn!("(Ignored) Internal error when indexing {}: {:?}", k, e);
                                                         }
                                                         events += 1;
```

Comment:
Actually, after submitting my review, *then* I noticed the `events` counter is still around... why not just check if `indexed` is empty instead?

Reply:
It wouldn't catch the 'deleted' events so using indexed is empty() would cause it to miss publishing snapshots of deleted documents.

Reply:
Speaking of that, I found myself wanting an event when an SBOM was deleted from the index as well as added -- i.e. I want to know the index has changed in any form or fashion. So `indexed` might hold structs with fields for `key` and `action` maybe -- denoting whether the key was added or removed -- and those same structs could be sent to the `indexed` topic.

Just a thought.

Reply:
Yeah, maybe renaming/repurposing it to sbom-changed along with key and event type.